### DISCUSSION TITLE
$ Setting changes NOT permanent

### DISCUSSION THREAD

**Discussion by massiveradiation** [grblHAL/core#583] [General]
Hi Folks, 
I'm just learning my way around grblHAL, coming from a grbl environment. In grbl, if I issue a setting change, like $20=0 in the console, that change is permanently set. In other words, when power is cycled on the board, the setting change remains. When I'm trying this from console in gsender with grblHAL, I can see that when I issue the command $20=0 and then list the settings with $$ command, it's there. But, after a power cycle, it goes back to what it was. 
What am I not getting? Is the only way to change a setting permanently to flash firmware on the board? Do I have to have an SD card installed on the board to retain settings?
Thanks in advance!

**Comment by terjeio** [comment]
Here is help output for `$20`:
```
$$=20
Enables soft limits checks within machine travel and sets alarm when exceeded. Requires homing.
```
If I attempt to enable soft limits with `$20=1` when `$22=0` I get this response:
`error:10`
which is translates to  "Soft limits cannot be enabled without homing also enabled."

So you will have to enable homing first by setting `$22` to a non-zero value. Note that soft limits will not be enforced for axes that are not homed.

> When I'm trying this from console in gsender with grblHAL, I can see that when I issue the command 20 = 0 a n d t h e n l i s t t h e s e t t i n g s w i t h $ command, it's there.

That could be due to a bug in gsender? I get 0 back from the controller in response to `$20` or `$$`.
